core/server/proxy/src/main/java/alluxio/proxy/s3/S3RestUtils.java

```diff
@@ -198,7 +200,11 @@ public static Response createErrorResponse(AlluxioStatusException e) {
       // 500
       s3ErrorCode = S3ErrorCode.INTERNAL_ERROR;
     }
-    S3Error errorResponse = new S3Error(e.getMessage(), s3ErrorCode);
+    if (StringUtils.isBlank(resource)) {
```

Comment:
TIL: `StringUtils.isBlank()` exists.

I think the Java-native approach is to overload the method signature though. See my other comment.